api/src/test/java/bio/terra/tanagra/testing/GeneratedSqlUtils.java

```diff
@@ -48,10 +48,39 @@ private GeneratedSqlUtils() {}
    */
   public static void checkMatchesOrOverwriteGoldenFile(String generatedSql, String fileName)
       throws IOException {
+    checkMatchesOrOverwriteGoldenFile(generatedSql, fileName, null);
+  }
+
+  /**
+   * This method is intended for SQL strings that include generated table aliases. This is the case
+   * for relationships that are defined by intermediate tables. If the relationshipName is not null,
+   * then before doing the comparison, this method replaces the generated table aliases in the
+   * expected SQL with those in the actual SQL. This is necessary because generated table names
```

Comment:
I think it would also be reasonable to inject a seeded Random from the test, if that's not already happening. This might be simpler.

Reply:
Good idea. I've done something similar before using an interface and annotating different implementation with different Spring profiles. Perhaps that approach will work here too. I'll plan to tackle that as a separate PR.